packages/docusaurus-theme-classic/src/theme/DocItem/index.js

```diff
@@ -14,7 +14,75 @@ import DocPaginator from '@theme/DocPaginator';
 
 import styles from './styles.module.css';
 
+let headersCache = [];
+let tableOfContentLinksCache = [];
+let lastActiveTableOfContentLink;
+
+function getActiveHeader() {
+  const TOP_OFFSET = 80;
```

Comment:
This should configurable as part of the hook arguments for cases where the navbar has a different height.